diff --git a/editablefields_as_link.module b/editablefields_as_link.module
index ee72322..f4e8776 100644
--- a/editablefields_as_link.module
+++ b/editablefields_as_link.module
@@ -200,6 +200,8 @@ function editablefields_as_link_ajax_edit($entity_type, $bundle, $entity_id, $en
     '#commands' => array(
       ajax_command_replace($link_id_selector . '__view', drupal_render($form)),
       ajax_command_remove($link_id_selector),
+      // Invoke script, which help to remove 'TypeError: validator is undefined' clientside module js error.
+      ajax_command_invoke(NULL, "assign_clienside_validation", array($form['wrapper_id']['#value'])),
     ),
   );
 
@@ -207,6 +209,13 @@ function editablefields_as_link_ajax_edit($entity_type, $bundle, $entity_id, $en
 }
 
 /**
+ * Javascript, which assign clientside validation rules to editing form.
+ */
+function editablefields_as_link_init() {
+  drupal_add_js(drupal_get_path('module', 'editablefields_as_link') . '/editablefields_as_link_clientside_validation.js');
+}
+
+/**
  * Form builder for the AJAX-link editable field form.
  */
 function editablefields_as_link_form($form, &$form_state, $entity_type, $entity, $field, $instance, $langcode, $display) {
diff --git a/editablefields_as_link_clientside_validation.js b/editablefields_as_link_clientside_validation.js
new file mode 100644
index 0000000..472929b
--- /dev/null
+++ b/editablefields_as_link_clientside_validation.js
@@ -0,0 +1,7 @@
+(function($) {
+  $.fn.assign_clienside_validation = function(wrapper) {
+    jQuery('#editablefields-as-link-form').validate({});
+    var processed_element = $('#'+wrapper).find('input,select').first();
+    $(wrapper).ajaxSend(function(){processed_element.blur()});
+  };
+})(jQuery);
\ No newline at end of file
