diff --git a/modules/system/system.admin.inc b/modules/system/system.admin.inc
index 05543be..df3a43f 100644
--- a/modules/system/system.admin.inc
+++ b/modules/system/system.admin.inc
@@ -2577,6 +2577,8 @@ function theme_status_report($variables) {
     if (empty($requirement['#type'])) {
       $severity = $severities[isset($requirement['severity']) ? (int) $requirement['severity'] : 0];
       $severity['icon'] = '<div title="' . $severity['title'] . '"><span class="element-invisible">' . $severity['title'] . '</span></div>';
+      // The requirement value is optional and so may need to be explicitly set.
+      //$requirement['value'] = isset($requirement['value']) ? $requirement['value'] : '';
 
       // Output table row(s)
       if (!empty($requirement['description'])) {
