diff --git a/field_collection.module b/field_collection.module
index 01dcc6c..9d8d656 100644
--- a/field_collection.module
+++ b/field_collection.module
@@ -1242,7 +1242,7 @@ function field_collection_field_widget_embed_validate($element, &$form_state, $c
 
   // Only if the form is being submitted, finish the collection entity and
   // prepare it for saving.
-  if ($form_state['submitted'] && !form_get_errors()) {
+  if ($form_state['submitted'] && !form_get_errors() && $form_state['clicked_button']['#name'] == 'op') {
 
     field_attach_submit('field_collection_item', $field_collection_item, $element, $form_state);
 
