diff --git a/geofield.module b/geofield.module
index 1b11870..fd184c4 100644
--- a/geofield.module
+++ b/geofield.module
@@ -198,7 +198,12 @@ function geofield_field_validate($entity_type, $entity, $field, $instance, $lang
  */
 function geofield_validate_geom($item) {
   try {  
-   $values = geofield_compute_values($item['geom'], $item['input_format']);
+   if (is_array($item)) {
+     $values = geofield_compute_values($item['geom'], $item['input_format']);
+   }
+   else {
+     $values = geofield_compute_values($item);
+   }
   }  
   catch (Exception $e) {  
    return $e;  
