diff --git a/i18n_field/i18n_field.module b/i18n_field/i18n_field.module
index 2d6bacd..86042c9 100644
--- a/i18n_field/i18n_field.module
+++ b/i18n_field/i18n_field.module
@@ -221,6 +221,7 @@ function i18n_field_field_storage_info_alter(&$info) {
  */
 function i18n_field_field_widget_form(&$form, &$form_state, $field, $instance, $langcode, $items, $delta, $element) {
   global $language;
+  $original_element = $element;
   // The field language may affect some variables (default) but not others (description will be in current page language)
   $i18n_langcode = empty($element['#language']) || $element['#language'] == LANGUAGE_NONE ? $language->language : $element['#language'];
   // Translate instance to current page language and set to form_state
@@ -247,6 +248,10 @@ function i18n_field_field_widget_form(&$form, &$form_state, $field, $instance, $
   // Redirect through original module_field_widget_form()
   if (!empty($instance['widget']['i18n_field_callbacks'])) {
     foreach ($instance['widget']['i18n_field_callbacks'] as $function) {
+      if (!$element) {
+        // If element is empty, use the origianl one.
+        $element = $original_element;
+      }
       $element = $function($form, $form_state, $field, $instance_current, $langcode, $items, $delta, $element);
     }
   }
