diff --git a/theme.inc b/theme.inc
index 71d4ffa..fa0bd3d 100644
--- a/theme.inc
+++ b/theme.inc
@@ -179,6 +179,13 @@ function theme_field_group_multiple_table_form($variables) {
 
     foreach (element_children($field_data, TRUE) as $field_name) {
       $data = $field_data[$field_name];
+      // Phone Number Support
+      $lang = array_shift(array_keys($data));
+      if (isset($data[$lang]['#type']) && $data[$lang]['#type'] == "phone_number" && isset($data[$lang]['#entity']->$field_name)) {
+        $default_values = $data[$lang]['#entity']->$field_name;
+        $data[$lang]['number']['#value'] = $default_values[$lang][$delta]['number'];
+        $data[$lang]['country_codes']['#value'] = $default_values[$lang][$delta]['country_codes'];
+      }
       $row[] = array("data" => drupal_render($data), "class" => "");
     }
     $rows[] = array("data" => $row, "class" => array("row-delta-" . $delta));
