diff --git a/plugins/views_data_export_plugin_style_export.inc b/plugins/views_data_export_plugin_style_export.inc
index 934fb48..60b64cb 100644
--- a/plugins/views_data_export_plugin_style_export.inc
+++ b/plugins/views_data_export_plugin_style_export.inc
@@ -311,7 +311,7 @@ class views_data_export_plugin_style_export extends views_plugin_style {
       $count = 0;
       foreach ($view->display_handler->get_handlers('argument') as $arg => $handler) {
         $token = '%' . ++$count;
-        $tokens[$token . '-title'] = check_plain($handler->title());
+        $tokens[$token . '-title'] = $handler->get_title();
         $tokens[$token . '-value'] = isset($view->args[$count - 1]) ? check_plain($view->args[$count - 1]) : '';
       }
 
